src/gas/mod.rs

```diff
@@ -250,7 +250,7 @@ fn add_grow_counter<R: Rules>(
 	let mut b = builder::from_module(module);
 	b.push_function(
 		builder::function()
-			.signature().params().i32().build().with_return_type(Some(elements::ValueType::I32)).build()
+			.signature().with_param(ValueType::I32).with_result(ValueType::I32).build()
```

Comment:
@pepyakin I was wondering why we even need a result here, I mean shouldn't the grow counter never return anything?

Reply:
TBH, I am a bit confused. Seal doesn't use (@athei correct me if I am wrong) a grow counter and I am not sure who does.

This grow counter looks to me a remnant from the times of parity-ethereum kovan contracts.

Reply:
> TBH, I am a bit confused. Seal doesn't use (@athei correct me if I am wrong) a grow counter and I am not sure who does.

Right now it does not use the grow counter (but it easily could). The memory.grow instruction is just benchmarked with its maximum argument (16k according to the default schedule). However, that can become infeasible when we want to increase this size. Then we might need the grow counter.

> @pepyakin I was wondering why we even need a result here, I mean shouldn't the grow counter never return anything?

Because the grow counter wraps the `memory.grow` instruction which returns an `i32`.